[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - 676191 - LOGS_CAPTURED - Started capturing logs in process (pid: 676191).
[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - 676191 - linear_regression - STEP_START - Started execution of step "linear_regression".
[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - linear_regression - Loading file from: /home/kai/Git/twai_mlops_assignment2-main/src/tmp9wf98lfe/storage/test_data using PickledObjectFilesystemIOManager...
[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - 676191 - linear_regression - LOADED_INPUT - Loaded input "test_data" using input manager "io_manager", from output "test_data" of step "train_test_data"
[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - linear_regression - Loading file from: /home/kai/Git/twai_mlops_assignment2-main/src/tmp9wf98lfe/storage/train_data using PickledObjectFilesystemIOManager...
[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - 676191 - linear_regression - LOADED_INPUT - Loaded input "train_data" using input manager "io_manager", from output "train_data" of step "train_test_data"
[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - 676191 - linear_regression - STEP_INPUT - Got input "test_data" of type "DataFrame". (Type check passed).
[32m2024-12-11 16:48:41 +0100[0m - dagster - [34mDEBUG[0m - __ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - 676191 - linear_regression - STEP_INPUT - Got input "train_data" of type "DataFrame". (Type check passed).
2024/12/11 16:48:41 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/mlflow/types/utils.py:407: UserWarning: Hint: Inferred schema contains integer column(s). Integer columns in Python cannot represent missing values. If your input data contains missing values at inference time, it will be encoded as floats and will cause a schema enforcement error. The best way to avoid this problem is to infer the model schema based on a realistic data sample (training dataset) that includes missing values. Alternatively, you can declare integer columns as doubles (float64) whenever these columns may have missing values. See `Handling Integers With Missing Values <https://www.mlflow.org/docs/latest/models.html#handling-integers-with-missing-values>`_ for more details."
[32m2024-12-11 16:48:57 +0100[0m - dagster - [34mERROR[0m - [31m__ASSET_JOB - bfb44330-9e86-42df-8e2f-7001265a5345 - 676191 - linear_regression - STEP_FAILURE - Execution of step "linear_regression" failed.

dagster._core.errors.DagsterExecutionStepExecutionError: Error occurred while executing op "linear_regression"::

sklearn.exceptions.NotFittedError: This ColumnTransformer instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator.

Stack Trace:
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/dagster/_core/execution/plan/utils.py", line 54, in op_execution_error_boundary
    yield
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/dagster/_utils/__init__.py", line 490, in iterate_with_context
    next_output = next(iterator)
                  ^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 140, in _coerce_op_compute_fn_to_iterator
    result = invoke_compute_fn(
             ^^^^^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 128, in invoke_compute_fn
    return fn(context, **args_to_pass) if context_arg_provided else fn(**args_to_pass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kai/Git/twai_mlops_assignment2-main/src/bike_share/assets/models.py", line 55, in linear_regression
    create_pipeline(LinearRegression()).score(test_input, test_output)
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/mlflow/utils/autologging_utils/safety.py", line 593, in safe_patch_function
    patch_function(call_original, *args, **kwargs)
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/mlflow/sklearn/__init__.py", line 1777, in patched_model_score
    return original(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/mlflow/utils/autologging_utils/safety.py", line 574, in call_original
    return call_original_fn_with_event_logging(_original_fn, og_args, og_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/mlflow/utils/autologging_utils/safety.py", line 509, in call_original_fn_with_event_logging
    original_fn_result = original_fn(*og_args, **og_kwargs)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/mlflow/utils/autologging_utils/safety.py", line 571, in _original_fn
    original_result = original(*_og_args, **_og_kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/sklearn/pipeline.py", line 1000, in score
    Xt = transform.transform(Xt)
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/sklearn/utils/_set_output.py", line 316, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/sklearn/compose/_column_transformer.py", line 1035, in transform
    check_is_fitted(self)
  File "/home/kai/miniconda3/envs/bike-share/lib/python3.12/site-packages/sklearn/utils/validation.py", line 1661, in check_is_fitted
    raise NotFittedError(msg % {"name": type(estimator).__name__})
[0m
